### ISSUE TITLE
net/http/header limits introduced in 0.4.0 are too low

### ISSUE THREAD

**Issue by dpsi** [ruby/net-http#173]
_Originally submitted as [https://bugs.ruby-lang.org/issues/20247](https://bugs.ruby-lang.org/issues/20247)_

Hello, some of my HTTP tests are failing on Ruby 3.3 due to ArgumentError too long. I am trying to update from Ruby 2.7. I did not see any mention of this change in the net/http changelog, but looking at git history, the limit was added between gem versions 0.3.2 and 0.4.0. https://github.com/ruby/ruby/commit/d8b8294c28a09278de357c26b291abf1b9f3cc5d

I send HTTP requests with long header keys in my testing suite. Many webservers such as apache, nginx, IIS permit the total length of HTTP headers to be 8K or longer. Tomcat for example allows 48K. I am trying to send a request with a header key length of 24K. I think the limit of 1K is too low.

There was not a clear reason for the change, so I request that it gets reverted.
Alternatively the limit could be configurable.

The code being affected by the limit was written for Ruby 1.8, so I am surprised such longstanding behaviour was removed without being listed in the changelog.

**Comment by hsbt** [comment]
>I think the limit of 1K is too low.

I don't think so. If you have an issue with real world problem without test case, I'll consider increasing that limit or add configurable feature.
